analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -0,0 +1,127 @@
+        const val YARN_PATH_PROPERTY_NAME = "yarnPath"
+    }
+
+    override fun hasLockFile(projectDir: File) = hasYarnLockFile(projectDir)
+
+    override fun command(workingDir: File?): String {
+        return workingDir?.let {
+            yarn2CommandsByPath.getOrPut(it) {
+                val yarnConfig = yamlMapper.readTree(workingDir.resolve(YARN2_RESOURCE_FILE))
+                val yarnCommand = yarnConfig.get(YARN_PATH_PROPERTY_NAME)
+                yarnCommand?.let { command ->
+                    val yarnExecutable = workingDir.resolve(command.textValue())
+                    // TODO: Yarn2 executable is a `cjs` file. Check if under Windows it needs to be run with `node`.
+
+                    // TODO: This is a security risk to blindly run code coming from a repository other than ORT's. ORT
+                    //  should download the Yarn2 binary from the official repository and run it.
+                    if (yarnExecutable.exists()) yarnExecutable.absolutePath else null
+                } ?: throw IllegalStateException("No Yarn executable could be found in 'yarnrc.yml'.")
+            }
+        } ?: "yarn2+ executable' in .yarn/releases/ and configured in .yarnrc.yml"
+    }
+
+    override fun getVersion(workingDir: File?): String =
+        // `getVersion` with a `null` parameter is called once by the Analyzer to get the version of the global tool.
+        // For Yarn2+, the version is specific to each definition file being scanned therefore a global version doesn't
+        // apply.
+        // TODO: An alternative would be to collate the versions of all tools in `yarn2CommandsByPath`.
+        if (workingDir == null) "N/A" else super.getVersion(workingDir)
+
+    override fun getVersionRequirement(): Requirement = Requirement.buildNPM(">=2.0.0")
+
+    override fun mapDefinitionFiles(definitionFiles: List<File>) = mapDefinitionFilesForYarn2(definitionFiles).toList()
+
+    override fun beforeResolution(definitionFiles: List<File>) =
+        // We depend on a version >= 2, so we check the version for safety.
+        definitionFiles.forEach { checkVersion(it.parentFile) }
+
+    override fun runInstall(workingDir: File): ProcessCapture =
+        // It should be enough for Yarn2+ to parse the lockfile to get the dependencies, and, if the former doesn't
+        // exist, run `yarn install --mode update-lockfile` to generate it (since yarn version 3.0.0-rc.10).
+        // Unfortunately, the lockfile doesn't contain the information required to build ORT's packages. An alternative
+        // could be to run `npm show --json` on each package present in the lockfile and rebuild a node_modules
+        // file hierarchy to be able to reuse the code present in the Npm package manager.
+        // Fortunately, an easier solution exist: by setting the option `nodeLinker` in the `yarnrc` file or as an
+        // environment variable, one can instruct Yarn to generate the node_modules files hierarchy.
```

Comment:
As the alternatives are not used in this code, how about removing this part of the comment?

Reply:
I wanted to explain why there is this env variable. Can we keep it ?